src/main/java/com/amazon/opendistroforelasticsearch/knn/index/MethodComponent.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.knn.index;
+
+import org.elasticsearch.common.ValidationException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * MethodComponent defines the structure of an individual component that can make up an index
+ */
+public class MethodComponent {
+
+    private String name;
+    private Map<String, Parameter<?>> parameters;
+
+    /**
+     * Constructor
+     *
+     * @param builder to build method component
+     */
+    private MethodComponent(Builder builder) {
```

Comment:
null check for builder

Reply:
My thinking is that because the component is built from the builder and is private, a null check is not needed. What do you think?

Reply:
Yes. You are right.